xapian-applications/omega/query.cc

```diff
@@ -2051,6 +2060,16 @@ eval(const string &fmt, const vector<string> &param)
 		}
 		break;
 	    }
+	    case CMD_random: {
+		if (args.size() == 1) {
+		    uniform_int_distribution<int>
+			distr(0, string_to_int(args[0]));
+		    value = str(distr(rng));
+		} else {
+		    throw "$random requires exactly 1 argument";
+		}
```

Comment:
You don't need to check the number of arguments here - `T(random,	   1, 1, N, 0), // return a random number` means that `$random` will only be passed one argument (the "1, 1" means "minimum 1, maximum 1").

Reply:
Fixed in the latest commit.